resolwe/flow/views.py

```diff
@@ -135,23 +135,35 @@ class ResolweCreateDataModelMixin(ResolweCreateModelMixin):
     """
     def create(self, request, *args, **kwargs):
         collections = request.data.get('collections', [])
-        if len(collections) != 1:
-            return Response({'collections': 'Exactly one id required on create.'},
-                            status=status.HTTP_400_BAD_REQUEST)
-
-        try:
-            collection = Collection.objects.get(pk=collections[0])
-        except Collection.DoesNotExist:
-            return Response({'collections': ['Invalid pk "{}" - object does not exist.'.format(collections[0])]},
-                            status=status.HTTP_400_BAD_REQUEST)
-
-        if not request.user.has_perm('add_collection', obj=collection):
-            if request.user.is_authenticated():
-                raise exceptions.PermissionDenied
-            else:
-                raise exceptions.NotFound
+        
+        for collection_id in collections:
```

Comment:
The web devs (@kostko, @hadalin) should tell how they would like to append data objects to collections at create and what they would want in return. Adding a data object to a one collection is definitely used in GenBoard. The zero collections alternative is used by @JureZmrzlikar in resolwe-bio-py. I do not expect a case with multiple collections in the near future. But I do not like exceptions and it is quite ugly to me to support just one element in an array.